core/sql/exp/exp_function.cpp

```diff
@@ -6503,8 +6503,19 @@ ex_expr::exp_return_type ex_function_json_object_field_text::eval(char *op_data[
         Int32 prec2 = ((SimpleType *)getOperand(2))->getPrecision();
         len2 = Attributes::trimFillerSpaces( op_data[2], prec2, len2, cs );
     }
+
     char *rltStr = NULL;
-    JsonReturnType ret = json_extract_path_text(&rltStr, op_data[1], 1, op_data[2]);
+    char *jsonStr = new(heap) char[len1+1];
+    char *jsonAttr = new(heap) char[len2+1];
+    if (jsonStr == NULL || jsonAttr == NULL)
+    {
+        return ex_expr::EXPR_ERROR;
+    }
+    memset(jsonStr, 0, len1+1);
+    memset(jsonAttr, 0, len2+1);
```

Comment:
Consider setting the null character after strncpy instead

Reply:
strncpy(jsonStr, op_data[1], len1);
jsonStr[len1] = '\0';
Do you mean the change should look like above?

Reply:
Yes. No need to do memset.